src/main/java/com/atlassian/db/replica/spi/circuitbreaker/CircuitBreaker.java

```diff
@@ -1,9 +1,18 @@
 package com.atlassian.db.replica.spi.circuitbreaker;
 
-import com.atlassian.db.replica.api.circuitbreaker.BreakerState;
+import com.atlassian.db.replica.api.SqlCall;
+
+import java.sql.SQLException;
 
 public interface CircuitBreaker {
-    BreakerState getState();
 
-    void handle(Throwable throwable);
+    boolean canCall();
+
+    <T> T handle(SqlCall<T> call) throws SQLException;
+
+    void handle(SqlRunnable runnable) throws SQLException;
+
+    interface SqlRunnable {
```

Comment:
Why inner? Doesn't deserve their own file?

Reply:
BTW why do we need it?

Reply:
It's moved from the old inner `BreakerHandler.SqlRunnable`. I will extract it.

I will also try to remove the need for it

Reply:
Cannot get rid of `SqlRunnable`:
![image](https://user-images.githubusercontent.com/1194704/106007738-e78ca280-60b6-11eb-939d-b7cb8526140a.png)

Standard split is `Callable`/`Runnable`.
Now we have `SqlCall`/`SqlRunnable`.
For consistency, we should go for:
1. short: `SqlCall`/`SqlRun`
2. idiomatic: `SqlCallable`/`SqlRunnable`

I'd default to *short*, but I will not defend that